File: external_apps/mobile_ios/mobile_ios/Core/Navigation/DeepLinkHandler.swift
//
//  DeepLinkHandler.swift
//  mobile_ios
//
//  Created for Deep Linking Feature
//

import Foundation
import SwiftUI

class DeepLinkHandler {
    static let shared = DeepLinkHandler()
    
    private init() {}
    
    func handleDeepLink(_ urlString: String?) -> String? {
        guard let urlString = urlString,
              let url = URL(string: urlString) else {
            return nil
        }
        
        switch url.host {
        case "profile":
            if let userId = url.queryParameters?["userId"] {
                return "profile/\(userId)"
            }
        case "chat":
            if let chatId = url.queryParameters?["chatId"] {
                return "chat/\(chatId)"
            }
        case "notification":
            if let notificationId = url.queryParameters?["id"] {
                return "notification/\(notificationId)"
            }
        case "payment":
            if let invoiceId = url.queryParameters?["invoiceId"] {
                return "payment/invoice/\(invoiceId)"
            }
        default:
            return "home"
        }
        
        return nil
    }
    
    // MARK: - Deep Link Builders
    static func buildProfileLink(userId: String) -> String {
        return "myapp://profile?userId=\(userId)"
    }
    
    static func buildChatLink(chatId: String) -> String {
        return "myapp://chat?chatId=\(chatId)"
    }
    
    static func buildNotificationLink(notificationId: String) -> String {
        return "myapp://notification?id=\(notificationId)"
    }
    
    static func buildPaymentLink(invoiceId: String) -> String {
        return "myapp://payment?invoiceId=\(invoiceId)"
    }
}

extension URL {
    var queryParameters: [String: String]? {
        guard let components = URLComponents(url: self, resolvingAgainstBaseURL: false),
              let queryItems = components.queryItems else {
            return nil
        }
        
        var parameters = [String: String]()
        for item in queryItems {
            parameters[item.name] = item.value
        }
        return parameters
    }
}

File: external_apps/mobile_android/app/src/main/java/com/example/mobile_android/data/notification/DeepLinkHandler.kt
package com.example.mobile_android.data.notification

import android.content.Intent
import android.net.Uri
import androidx.navigation.NavController
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class DeepLinkHandler @Inject constructor() {
    
    fun handleDeepLink(deepLink: String?, navController: NavController) {
        if (deepLink.isNullOrEmpty()) return
        
        try {
            val uri = Uri.parse(deepLink)
            when (uri.host) {
                "profile" -> {
                    val userId = uri.getQueryParameter("userId")
                    navController.navigate("profile/$userId")
                }
                "chat" -> {
                    val chatId = uri.getQueryParameter("chatId")
                    navController.navigate("chat/$chatId")
                }
                "notification" -> {
                    val notificationId = uri.getQueryParameter("id")
                    navController.navigate("notification/$notificationId")
                }
                "payment" -> {
                    val invoiceId = uri.getQueryParameter("invoiceId")
                    navController.navigate("payment/invoice/$invoiceId")
                }
                else -> {
                    // Navigate to home or handle unknown deep link
                    navController.navigate("home")
                }
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
    
    fun handleIntent(intent: Intent?, navController: NavController) {
        intent?.getStringExtra("deepLink")?.let { deepLink ->
            handleDeepLink(deepLink, navController)
        }
    }
    
    companion object {
        // Deep link builders
        fun buildProfileLink(userId: String): String {
            return "myapp://profile?userId=$userId"
        }
        
        fun buildChatLink(chatId: String): String {
            return "myapp://chat?chatId=$chatId"
        }
        
        fun buildNotificationLink(notificationId: String): String {
            return "myapp://notification?id=$notificationId"
        }
        
        fun buildPaymentLink(invoiceId: String): String {
            return "myapp://payment?invoiceId=$invoiceId"
        }
    }
}
